### PR TITLE
Migrate from AsssertJ to AssertK

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

**Review by JakeWharton — APPROVED** [review]
Lots of the `!!` could be replaced with inline `isNotNull()` assertions, but it doesn't really matter all that much.

File: wire-compiler/src/test/java/com/squareup/wire/ManifestParseTest.kt [hunk 1]
```diff
@@ -69,7 +72,7 @@ class ManifestParseTest {
     """.trimMargin()
 
     val modules = parseManifestModules(yaml)
-    assertThat(modules.keys).containsExactly("one", "two")
-    assertThat(modules.getValue("one").dependencies).containsExactly("two")
+    assertThat(modules.keys.toList()).containsExactly("one", "two")
```
[the remark this answers is not in the document]
Comment (oldergod) [reply wire-compiler/src/test/java/com/squareup/wire/ManifestParseTest.kt:75]: lmao

[the remark this answers is not in the document]
Comment (oldergod) [reply wire-compiler/src/test/java/com/squareup/wire/ManifestParseTest.kt:75]: But assertJ containsExactly also checks the order, which is why I keep this assertion here.

File: wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/InteropChecker.kt [hunk 2]
```diff
@@ -107,12 +109,12 @@ class InteropChecker(
...
-        |Expected :$message (unknownFields: `${(message as com.squareup.wire.Message<*, *>).unknownFields.hex()}`)
-        |Actual   :$fromJson (unknownFields: `${(fromJson as com.squareup.wire.Message<*, *>).unknownFields.hex()}`)
-        """.trimMargin()
-      }
+      // .overridingErrorMessage {
+      //   """
+      //   |Expected :$message (unknownFields: `${(message as com.squareup.wire.Message<*, *>).unknownFields.hex()}`)
+      //   |Actual   :$fromJson (unknownFields: `${(fromJson as com.squareup.wire.Message<*, *>).unknownFields.hex()}`)
+      //   """.trimMargin()
+      // }
```
[the remark this answers is not in the document]
Comment (oldergod) [reply wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/InteropChecker.kt:117]: very nice

File: wire-compiler/src/test/java/com/squareup/wire/ManifestParseTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (oldergod) [reply wire-compiler/src/test/java/com/squareup/wire/ManifestParseTest.kt:75]: but sets don't have ordering... so

**Comment by oldergod** [comment]
I addressed all comments, thank you

**Pull request merged**